src/Facade/Services/WarehouseFacadeService.cs

```diff
@@ -23,5 +29,82 @@ public IResult<MessageResult> MovePalletToUpper(int palletNumber, string referen
+                return new BadRequestResult<MessageResult>("Not a valid pallet number");
+            }
+
+            if (resource.Priority < 0 || resource.Priority > 100)
+            {
+                return new BadRequestResult<MessageResult>("Not a valid priority");
+            }
+
+            var employee = this.employeeRepository.FindById(resource.EmployeeId);
+            if (employee == null)
+            {
+                return new NotFoundResult<MessageResult>("Employee not found");
+            }
+
+            if (resource.TaskType == "MV")
+            {
+                if (this.warehouseService.MovePallet(resource.PalletNumber, resource.Destination, resource.Priority, employee))
+                {
+                    return new SuccessResult<MessageResult>(new MessageResult($"Task submitted to move pallet {resource.PalletNumber} to {resource.Destination}"));
+                }
+            }
+            else if (resource.TaskType == "AT")
+            {
+                if (this.warehouseService.AtMovePallet(
+                        resource.PalletNumber,
+                        resource.OriginalLocation,
+                        resource.Destination,
+                        resource.Priority,
+                        employee))
+                {
+                    return new SuccessResult<MessageResult>(
+                        new MessageResult(
+                            $"Task submitted to move pallet {resource.PalletNumber} from {resource.OriginalLocation} to {resource.Destination}"));
+                }
+            }
+            else if (resource.TaskType == "EM")
+            {
+                if (this.warehouseService.EmptyLocation(resource.PalletNumber, resource.OriginalLocation, resource.Priority, employee))
+                {
+                    return new SuccessResult<MessageResult>(new MessageResult($"Task submitted to move pallet {resource.PalletNumber} to {resource.Destination}"));
+                }
+            }
+
+            return new ServerFailureResult<MessageResult>("Failed to make task");
+        }
+
+        public IResult<WarehouseLocation> GetPalletLocation(int palletNumber)
+        {
+            var location = this.warehouseService.GetWarehouseLocation(string.Empty, palletNumber);
+            if (string.IsNullOrEmpty(location?.Location))
+            {
+                return new NotFoundResult<WarehouseLocation>("Pallet not on system");
+            }
+
+            return new SuccessResult<WarehouseLocation>(location);
+        }
+
+        public IResult<WarehouseLocation> GetPalletAtLocation(string location)
+        {
+            var warehouseLocation = this.warehouseService.GetWarehouseLocation(location, null);
+            if (warehouseLocation?.PalletId == null)
+            {
+                return new NotFoundResult<WarehouseLocation>("Pallet not on system");
+            }
+
+            return new SuccessResult<WarehouseLocation>(warehouseLocation);
+        }
```

Comment:
some of this feels a bit domain logic-y to me... although I'm not familair with the domain at all

Reply:
Nearly all the critical domain logic is down in the WCS_PACK and whatever replaces it.  There is checking of some parameters to determine if it is a valid request such as whether is it likely to be a valid priority or pallet number which does slightly feel a bit domain-y.  In reality the WCS_PACK or the call to the SCS PLC will come back with an error if you submit an invalid parameter so this is a bit of pre-optimization to not submit something to them that you know won't ever work.   So it might be stripped out at some point when this is enhanced to not merely say a success/fail on submitting a task but more detailed explanation for failure.